### ISSUE TITLE
Blue circles and line become gray and unresponsive 

### ISSUE THREAD

**Issue by jlazovskis** [rivetTDA/rivet#133]
The blue circles and the adjustable line become unresponsive sometimes. 

I loaded the file "senate104_RIVET_Codensity20thPercentile.txt" with homology 0, 40 x-bins and 40 y-bins. Then in the "Line Selection Window" when I drag the blue circles on the window boundary, both circles and the line become gray if the left endpoint is at the origin and the right endpoint is dragged right-to-left across the top-left corner of the diagram (I can sometimes reproduce the issue by also dragging the left endpoint left-to-right towards the origin). Then the window looks as in the screenshot below.

![issue1](https://user-images.githubusercontent.com/14241342/44291882-f1075480-a246-11e8-9fd8-67b3d76acc64.png)

Eveything else works - I can click on the bars, hovering over regions shows the dimension, the bottom options are accessible, I just can't move the main line or its endpoints. By clicking on the "Angle" or "Offset" up or down directions I can make the gray circle blue again, but the graph looks jumbled, as in the screenshot below.

![issue2](https://user-images.githubusercontent.com/14241342/44292093-66bff000-a248-11e8-9449-66f78445ee4e.png)

As soon as I drag either of the blue circles again, everything goes back to normal.

I cloned Rivet from this repository on the morning of 2018-08-15. My system is as follows:

- Linux Ubuntu 17.10
- Intel Core i7-4510U CPU @ 2.00 GHz
- RIVET 0.4
- CMake version 3.9.1
- QMake version 3.1
- Qt version 5.9.1
- Boost version 1.62

**Comment by mlesnick** [comment]
Jānis, thank you for this careful bug report.  We will have a look.
I'm not able to reproduce this on my OS X build, but I will see next week
if I can reproduce it on my Linux machine.

**Comment by SimonSegert** [comment]
I was able to reproduce this on my Mac. Commit 63e8bb0 seemed to resolve the issue for me.

**Comment by mlwright84** [comment]
Thanks, Simon, for your work on this. Unfortunately, I'm still experiencing this problem, even with your recent commit. Every time I load the senate104 data set (H_0, 40x40 bins) and move the upper control dot to the left, the line turns grey. I'm using Ubuntu 18.04, with qmake 3.1, Qt 5.9.5, and gcc Ubuntu 7.3.0-16ubuntu3.

**Comment by SimonSegert** [comment]
Matthew, I’m using Qt Creator 4.5.0, with Qt 5.10.0 and Clang 7.0. Perhaps it has to do with your version of Qt?

**Comment by mlwright84** [comment]
Nevermind; my mistake. I forgot that we changed the name of the executable. Out of old habit, I was running RIVET instead of rivet_GUI. The issue appears to be fixed!

**Comment by mlesnick** [comment]
Thank you Simon and Matthew.  Jānis, if you find that the fix Simon pushed
works on your system, feel free to close the issue.

**Comment by jlazovskis** [comment]
Using Simon's commit (63e8bb0), everything works! So I suppose this is indeed closed, thanks for your work on this.

However (not sure if I should be posting this here or somewhere else) - using the latest commit (1534e8a), I get a different error. After running `./rivet_GUI` in the rivet directory and loading the same file with the same parameters, the progress bar window appears, but as soon as it reaches 100% it crashes, with the last few lines from the terminal as below.

```
console:  PROGRESS 30489

console:  PROGRESS 30490

console:  BARCODE TEMPLATE COMPUTATION COMPLETE: path traversal and persistence updates took 1270 milliseconds 

console:     building the line arrangement and computing all barcode templates took 2058 milliseconds 

console:  ARRANGEMENT: 

Loading file 
terminate called after throwing an instance of 'std::runtime_error'
  what():  Couldn't open  for reading
Aborted (core dumped)
```

I tried using `./rivet_console` with the same file and parameters, which seemed to run fine, but it didn't write to any file, terminating with simply `Wrote arrangement to`. Running the same command in Simon's commit the output file is created, terminating with 

```
Wrote arrangement to out.mif
Writing file: out.mif 
```

**Comment by mlwright84** [comment]
Thanks, Jānis. It looks like I broke something in my most recent commit.
I'll try to fix it now.

**Comment by mlwright84** [comment]
Jānis, I think fixed the most recent problem in Commit 4dd01de. Please try again and let us know if you encounter further problems. Thanks!
